### PR TITLE
Fix two-qubit decomp to consider arbitrary number of unitaries

### PR DESCRIPTION
**Pull request by glassnotes**
**Context:** Circuits with >3 two-qubit unitaries were yielding a `RecursionError` when put through the `unitary_to_rot` transform.

**Description of the Change:** Fixes the queuing in the `two_qubit_decomposition` method and adds a relevant unit test.

**Benefits:** Arbitrary number of two-qubit unitaries can now be decomposed.

**Possible Drawbacks:** None; although, I'm admittedly not sure why this fixed the problem.

**Related GitHub Issues:** #2014

### CODE REVIEW

File: pennylane/transforms/decompositions/two_qubit_unitary.py
Comment (josh146) [inline pennylane/transforms/decompositions/two_qubit_unitary.py:616]: This is the only change?

🤔 🤔 🤔 🤔 🤔

Comment (josh146) [reply pennylane/transforms/decompositions/two_qubit_unitary.py:616]: Hmm, the logic of the `qml.apply` function is quite different if `context` is passed!

```python
if hasattr(op, "queue"):
    # operator provides its own logic for queuing
    op.queue(context=context)
else:
    # append the operator directly to the relevant queuing context
    context.append(op)
```

My guess is that, in the old version, the `with current tape:` was causing the issue somehow?

**Review by josh146 — APPROVED** [review]
Thanks @glassnotes! Should we merge this into `v0.20.0-rc0` instead? v0.20 is due to be released this Tuesday :)

File: pennylane/transforms/decompositions/two_qubit_unitary.py
Comment (glassnotes) [reply pennylane/transforms/decompositions/two_qubit_unitary.py:616]: > This is the only change?

I know right? It's almost concerning :fearful: I was thinking in the old version maybe the contexts get weirdly nested somehow. Like we were already in a context, and then were doing, 'within the same context, do more stuff', and after a sufficient number of instances of that we hit the limit.

**Comment by glassnotes** [comment]
> Thanks @glassnotes! Should we merge this into v0.20.0-rc0 instead? v0.20 is due to be released this Tuesday :)

Your call - it's not a critical bug that will affect a lot of things, so I think it'd be fine to wait until the next release, but I'm good either way!

**Comment by josh146** [comment]
>  so I think it'd be fine to wait until the next release,

Okay, lets leave it for 0.21! More time for it to be tested as well.

**Pull request merged**
